vaas/vaas/metrics/prometheus.py

```diff
@@ -20,16 +20,25 @@ def __init__(self) -> None:
         self.metrics_bucket: Dict[str, Kind] = {}
         self.registry: CollectorRegistry = CollectorRegistry()
 
-    def get_or_create(self, name: str, kind: Type[Kind]) -> Kind:
-        metric: Optional[Kind] = self.metrics_bucket.get(name)
+    def get_or_create(self, full_name: str, kind: Type[Kind]) -> Kind:
+        metric: Optional[Kind] = self.metrics_bucket.get(full_name)
         if not metric:
-            if self.labels:
-                new_metrics: Kind = kind(name, name, self.labels.keys(), registry=self.registry).labels(**self.labels)
+            name, labels = self._split_name_and_labels(full_name)
+            if labels:
+                new_metrics: Kind = kind(name, name, labels.keys(), registry=self.registry).labels(**labels)
             else:
                 new_metrics: Kind = kind(name, name, registry=self.registry)
-            self.metrics_bucket[name] = new_metrics
-            return new_metrics
-        return metric
+            self.metrics_bucket[full_name] = new_metrics
+        return self.metrics_bucket[full_name]
+
+    def _split_name_and_labels(self, name: str) -> Tuple[str, Dict[str, str]]:
```

Comment:
1. Please give a brief comment explanation of what this help function does.
2. Maybe name of this function also should be more precise, like: _combine_metrics_name_and_labels.
3. If not maybe better to split responsibility for split and combine labels in separate functions because now this one function does both.

Reply:
renamed and explained. It should not be separated because:
- the logic and purpose is obvious
- we always need to fetch `name` and `labels` together - so there is no reason to parse the dotted name twice